pkgs/development/python-modules/django-q/default.nix

```diff
@@ -1,24 +1,43 @@
-{ lib, buildPythonPackage, fetchPypi, django-picklefield, arrow
-, blessed, django, future }:
+{ arrow
+, blessed
+, buildPythonPackage
+, django
+, django-picklefield
+, fetchFromGitHub
+, future
+, lib
+, poetry-core
+}:
 
 buildPythonPackage rec {
   pname = "django-q";
   version = "1.3.9";
+  format = "pyproject";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "5c6b4d530aa3aabf9c6aa57376da1ca2abf89a1562b77038b7a04e52a4a0a91b";
+  src = fetchFromGitHub {
+    owner = "Koed00";
+    repo = "django-q";
+    sha256 = "sha256-gFSrAl3QGoJEJfvTTvLQgViPPjeJ6BfvgEwgLLo+uAA=";
+    rev = "v${version}";
   };
 
+  nativeBuildInputs = [ poetry-core ];
+
   propagatedBuildInputs = [
-    django-picklefield arrow blessed django future
+    django-picklefield
+    arrow
+    blessed
+    django
+    future
   ];
 
+  # all checks need a running redis instance
   doCheck = false;
```

Comment:
Adding a `pythonImportsCheck` would be better than nothing.

Reply:
Also FWIW, we actually _can_ run a redis server for the `checkPhase`, see how `django-redis` does it.

Reply:
good idea, thanks! I added the importCheck and the redis server to the checks